### PR TITLE
[BrightcoveNew] Add playlist support

### PR DESCRIPTION
**Pull request by kylepw**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

I added support for [playlist streams](https://support.brightcove.com/implementing-playlists) to BrightcoveNewIE, including an additional test. Only differences between 'playlist' and 'video' streams are the 'xxxId' part of the URL and the format of the JSON response, both of which are covered in this PR.

### CODE REVIEW

**Comment by kylepw** [comment]
@dstftw Travis CI isn't building for some reason?

**Comment by kylepw** [comment]
@dstftw Hey, any chance I could get this reviewed? If this is accepted, then I can delegate my YahooJapansNews PR to Brightcove like you told me to. Thanks in advance!

**Comment by kylepw** [comment]
@dstftw Could you take a look at this? It's not much new code. I'd really like to add BC playlist support. Thanks!

**Comment by dstftw** [comment]
Does not seem to work.
```
youtube-dl https://players.brightcove.net/1752604059001/S13cJdUBz_default/index.html?playlistId=5718313430001
```
must extract playlist of 7 video entries not a single video with all playlist entries dumped as formats.

**Comment by kylepw** [comment]
> Does not seem to work.
> 
> ```
> youtube-dl https://players.brightcove.net/1752604059001/S13cJdUBz_default/index.html?playlistId=5718313430001
> ```
> 
> must extract playlist of 7 video entries not a single video with all playlist entries dumped as formats.

@dstftw Thanks for feedback. I implemented extraction in playlist format. Please confirm.

**Comment by kylepw** [comment]
> Does not seem to work.
> 
> ```
> youtube-dl https://players.brightcove.net/1752604059001/S13cJdUBz_default/index.html?playlistId=5718313430001
> ```
> 
> must extract playlist of 7 video entries not a single video with all playlist entries dumped as formats.

Okay, it extracts playlists now. How is this?

**Comment by kylepw** [comment]
> must extract playlist of 7 video entries not a single video with all playlist entries dumped as formats.

@dstftw Okay, it extracts playlists now. How is this?

File: youtube_dl/extractor/brightcove.py [hunk 1]
```diff
@@ -708,14 +721,25 @@ def build_format_id(kind):
...
         }
 
+    def _extract_playlist(self, json_data, headers):
+        playlist_id = json_data.get('id')
+        playlist_title = json_data.get('name')
+        playlist_description = json_data.get('description')
+        playlist_vids = json_data.get('videos', [])
+
+        return self.playlist_result(
+            [self._parse_brightcove_metadata(vid, vid.get('id'), headers) for vid in playlist_vids],
```
Comment (dstftw) [inline youtube_dl/extractor/brightcove.py]: Entries with invalid `vid.get('id')` should be skipped.

File: youtube_dl/extractor/brightcove.py [hunk 2]
```diff
@@ -516,6 +516,19 @@ class BrightcoveNewIE(AdobePassIE):
...
+        'info_dict': {
+            'id': '5743160747001',
+            'ext': 'mp4',
+            'title': 'prod ntv',
+            'uploader_id': '5690807595001',
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        }
```
Comment (dstftw) [inline youtube_dl/extractor/brightcove.py]: Make this test `only_matching`.

File: youtube_dl/extractor/brightcove.py [hunk 3]
```diff
@@ -708,14 +721,25 @@ def build_format_id(kind):
             'is_live': is_live,
         }
 
+    def _extract_playlist(self, json_data, headers):
```
Comment (dstftw) [inline youtube_dl/extractor/brightcove.py]: There is no point in a separate method.

File: youtube_dl/extractor/brightcove.py [hunk 4]
```diff
@@ -708,14 +721,25 @@ def build_format_id(kind):
...
+        playlist_id = json_data.get('id')
+        playlist_title = json_data.get('name')
+        playlist_description = json_data.get('description')
+        playlist_vids = json_data.get('videos', [])
```
Comment (dstftw) [inline youtube_dl/extractor/brightcove.py]: Inline all these.

File: youtube_dl/extractor/brightcove.py (hunk 2 above)
Comment (dstftw) [inline youtube_dl/extractor/brightcove.py]: Add the test from my post and test for a number of entries.

**Comment by kylepw** [comment]
@dstftw I made the requested changes. Please confirm.

**Pull request merged**